pkg/operator/k8sutil/node.go

```diff
@@ -263,8 +263,8 @@ func RookNodesMatchingKubernetesNodes(rookStorage rookalpha.StorageScopeSpec, ku
 			if rookNodeMatchesKubernetesNode(rn, kn) {
 				rn.Name = normalizeHostname(kn)
 
-				// modify the node Location if topology awareness enabled
-				if rookStorage.TopologyAware == true {
+				// modify the node location when running OSD on PVC
+				if len(rookStorage.StorageClassDeviceSets) != 0 {
```

Comment:
If we're removing the topologyAware property, i think we always need to execute `nodeTopologyLocation()` on the next line. If we find the properties on the nodes for zones/regions, we should just always apply them to the CRUSH map. It doesn't just apply to running OSDs on PVs. If OSDs are on raw devices the additional CRUSH hierarchy is going to be a benefit as well.